File: CarRental.DAL/Models/Booking.cs
namespace CarRental.Database.Models
{
    public class Booking
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public int CarId { get; set; }
        public int PickUpRegistrationId { get; set; }
        public int? ReturnRegistrationId { get; set; }

        public Customer Customer { get; set; }
        public Car Car { get; set; }
        public Registration PickUpRegistration { get; set; }
        public Registration ReturnRegistration { get; set; }
    }
}

File: CarRental.Web/ClientApp/src/app/models/booking.ts
import { Car } from './car';
import { Customer } from './customer';
import { Registration } from './registration';

export class Booking {
  id: number;
  customer: Customer;
  car: Car;
  pickUpRegistration: Registration;
  returnRegistration: Registration;

  constructor(customer: Customer, car: Car, pickUpRegistration: Registration, returnRegistration: Registration) {
    this.customer = customer;
    this.car = car;
    this.pickUpRegistration = pickUpRegistration;
    this.returnRegistration = returnRegistration;
  }
}
